framework/test_app/runners/k8s/k8s_base_runner.py

```diff
@@ -299,6 +299,43 @@ def _reuse_namespace(self) -> k8s.V1Namespace:
         logger.info("Reusing namespace: %s", self.k8s_namespace.name)
         return self.k8s_namespace.get()
 
+    def _create_pod_monitoring(
+        self,
+        template,
+        *,
+        namespace_name: str,
+        deployment_id: str,
+        pod_monitoring_name: str,
+        **kwargs,
+    ) -> k8s.PodMonitoring:
+        pod_monitoring = self._create_from_template(
+            template,
+            custom_object=True,
+            namespace_name=namespace_name,
+            deployment_id=deployment_id,
+            pod_monitoring_name=pod_monitoring_name,
+            **kwargs,
+        )
+        if not (
+            isinstance(pod_monitoring, k8s.PodMonitoring)
+            and pod_monitoring.kind == "PodMonitoring"
+        ):
+            raise _RunnerError(
+                f"Expected ResourceInstance[PodMonitoring] to be created from"
+                f" manifest {template}"
+            )
+        if pod_monitoring.metadata.namespace != namespace_name:
+            raise _RunnerError(
+                "PodMonitoring resource created with unexpected namespace: "
+                f"{pod_monitoring.metadata.namespace}"
+            )
```

Comment:
```suggestion
```

No need for this check: something should go horribly wrong for the namespace field to not apply.